presto-tablestore/src/test/java/com/facebook/presto/tablestore/TestTablestoreFacade.java

```diff
@@ -0,0 +1,1105 @@
+        assertEquals(GREATER_EQUAL, sf.getOperator());
+
+        r = Range.greaterThan(INTEGER, 10L); // (10,+∞)
+        res = transformFilterForColumnV2(PK1_INT, r);
+        sf = (SingleColumnValueRegexFilter) res;
+        assertNotNull(sf);
+        assertEquals(PK1_INT, sf.getColumnName());
+        assertEquals(10L, sf.getColumnValue().asLong());
+        assertEquals(GREATER_THAN, sf.getOperator());
+
+        r = Range.range(INTEGER, 1L, true, 2L, false); //[1,2)
+        res = transformFilterForColumnV2(PK1_INT, r);
+        CompositeColumnValueFilter cf = (CompositeColumnValueFilter) res;
+        assertNotNull(cf);
+        assertEquals(LogicOperator.AND, cf.getOperationType());
+        assertEquals(2, cf.getSubFilters().size());
+        SingleColumnValueRegexFilter sf0 = (SingleColumnValueRegexFilter) cf.getSubFilters().get(0);
+        SingleColumnValueRegexFilter sf1 = (SingleColumnValueRegexFilter) cf.getSubFilters().get(1);
+
+        assertEquals(PK1_INT, sf0.getColumnName());
+        assertEquals(1L, sf0.getColumnValue().asLong());
+        assertEquals(GREATER_EQUAL, sf0.getOperator());
+
+        assertEquals(PK1_INT, sf1.getColumnName());
+        assertEquals(2L, sf1.getColumnValue().asLong());
+        assertEquals(LESS_THAN, sf1.getOperator());
+    }
+
+    /**
+     * 类似于：0:(min,0) 1:(0,1) 2:(1,2) 3:(2,3) 4:(3,4) 5:(4,5) 6:(5,6) 7:(6,7) 8:(7,8) 9:(8,max)，第一个是分区，第二个是min，第三个是max
+     */
+    private String print(List<Split> rList)
+    {
+        return rList.stream().map(s -> {
+            PrimaryKey lower = s.getLowerBound();
+            PrimaryKey upper = s.getUpperBound();
+            String x = s.getLocation() + ":(" + (lower.equals(MIN_PK) ? "min" : lower.getPrimaryKeyColumn(0).getValue().asLong()) + ",";
+            x += (upper.equals(MAX_PK) ? "max" : upper.getPrimaryKeyColumn(0).getValue().asLong()) + ")";
+            return x;
+        }).reduce((a, b) -> a + " " + b).orElseThrow(NullPointerException::new);
+    }
+
+    private List<Split> mockSplitList(int size, int numberPerLocation)
+    {
+        List<Split> list = new ArrayList<>();
+        PrimaryKey lower = MIN_PK;
+        for (int i = 0; i < size; i++) {
+            Split s = new Split();
+            s.setLocation("" + (i / numberPerLocation));
+            PrimaryKey upper;
+            if (i == size - 1) {
+                upper = MAX_PK;
+            }
+            else {
+                upper = pk(fromLong(i));
+            }
+            s.setLowerBound(lower);
+            s.setUpperBound(upper);
+            list.add(s);
+            lower = upper;
+        }
+        return list;
+    }
+
+    @Test
+    public void testDivideAndMergeAndShuffle()
+    {
+        List<Split> list = mockSplitList(10, 3);
+        List<Split> rList = divideAndMergeAndShuffle(4, list);
+        assertEquals("0:(min,3) 1:(3,5) 2:(5,7) 2:(7,max)", print(rList));
+
+        //希望合并到3份，但也只能合并到4份，因为有些合并不了了
```

Comment:
Maybe remove the comment here? I guess no matter we merge them to 3 or 4, it won't influence the major features.

Reply:
done